### PR TITLE
[Overlapping Strategy] New form design (Create, Edit, Withdraw, Deposit)

### LINKED ISSUE #948: Overlapping - Edit price flow
# Edit overlapping strategy
as a user, i would like to have the option to edit the prices of an existing strategy.

`HINT: this design is planned to be included in the create, deposit and withdraw flows. more details in issue #1131.
`
## Mockups
[general flow designs](https://www.figma.com/file/YpsatcY9jtxJr0SJwlFYNj/Carbon-Screen-Design?type=design&node-id=14733-65980&mode=design&t=jzlsYBe0xiJDILJf-4)
[progress bar animation](https://www.figma.com/file/YpsatcY9jtxJr0SJwlFYNj/Carbon-Screen-Design?type=design&node-id=14733-69542&mode=design&t=jzlsYBe0xiJDILJf-4)

### Strategy menu
expose the option to "edit prices" for overlapping strategy

### Budget section
Edit price design is very similar to the create strategy page. given that, it has 3 sections:
1. price range - interactive chart + price input fields
2. spread - input of spread settings
3. budget - input of budget
This issue will address changes that are relevant to the `budget` section.
`how to expose parts in this section is explained below`

#### Select token
This section will allow users to choose which of the two tokens (base or quote) that are part of the strategy, they would like to interact with.
- default state - nothing is selected
- token positions - left token should be the base token. right token should be the quote token
- once a token is selected, user can only switch between the two token options and one is always selected

IMPORTANT: once token is selected, step 3 (distribution) should take the currently allocated budget for this token as if this is the user input value and recalculate the required budget for the other token (deposit/withdraw) based on the new price/spread/marginal price (as we do today).

Error message:
there is a single error message in this section that should be visible under the following condition:
IF user update price or spread AND no token was selected, show the error.
[mockup](https://www.figma.com/file/YpsatcY9jtxJr0SJwlFYNj/Carbon-Screen-Design?type=design&node-id=14733-68135&mode=design&t=X0hrFXBNX04yQUIA-4)

#### Edit budget
This section is optional and allow user to indicate a change in allocated budget. Basically, this will allow editing budgets as well during this flow.

- default state - closed. 
- on click - open the section and expose the additional fields (another click would close it again). if user input any value and close the section, delete the information so it wont appear on next open. 
- action selection - deposit/withdraw
-- default selected state - deposit option
-- when set to **deposit** - any input value will indicate the amount taken from the wallet and deposited into the strategy. this is identical to the **deposit flow** we currently have.
-- when set to **withdraw** - any input value will indicate the amount removed from the strategy and transferred to the wallet. this is identical to the **withdraw flow** we currently have.
- input value - based on the **action** selection, apply the deposit or withdraw input fields. Deposit field will show the max option with wallet balance, Withdraw field will show the max option with strategy budget.

IMPORTANT: user can leave the input field empty with no error/warning shown. once action and amount are selected, **distribution** should be updated accordingly (as we do today)

Error/warning messages:
any error/warning messages we currently have during this flow (insufficient balance for example), should be visible below the input field in this section.

#### Distribution
This section is for visibility only (user cannot interact with anything on this section).
- top bar is always representing the "sell" budget. bottom bar is always representing the "buy" budget.
- each order has a bar indicating the distribution between the **allocated** and **wallet** budgets.
```
example:
allocated = 100 tokens
wallet = 500 tokens
bar display 1/6 allocation (50% opacity) and 5/6 wallet (20% opacity)
how to fetch and calculate this?
1. fetch strategy balance of base and quote tokens
2. fetch wallet balance of base and quote tokens
3. allocation portion = (strategy.balance / (strategy.balance + wallet.balance))*100
```
- strategy portion is always in color with no opacity. wallet portion is always with opacity
- representing a change in allocation (following a budget recalculation flow)
-- IF **withdraw** amount > 0 or null
--- add arrow on the right end of the strategy allocation portion in the progress bar (pointing to the left)
--- identify the portion delta that should change 
--- Example: 
allocation 500 portion 50%
wallet 500 portion 50%
withdraw input amount = 200
current allocation = (500/(500+500))*100 = 50%
new allocation = (300/(300+700))*100 = 30%
split area delta = 50% - 30% = 20%
--- split it out of the bar (adding left and right borders/gap)
--- in a span of 1.5 seconds, empty this area from color from right to left (same color as the wallet portion) + move the arrows as the amount empty out until the left most arrow reach the right end of the allocation portion.
--- at the end of the animation, the arrows should be removed from view
-- IF **deposit** amount > 0 or null
--- add arrow on the right end of the strategy allocation portion in the progress bar (pointing to the right)
--- identify the portion delta that should change
--- Example:
allocation 100 portion 10%
wallet 900 portion 90%
deposit input amount = 200
current allocation = (100/(100+900))*100 = 10%
new allocation = (300/(300+700))*100 = 30%
split area delta = 30% - 10% = 20%
--- split it out of the bar (adding left and right borders/gap)
--- in a span of 1.5 seconds, fill this area with color from left to right (same color as the allocation portion) + move the arrows as the amount fills until the right most arrow reach the right end of the allocation portion.
--- at the end of the animation, the arrows should be removed from view
- supporting text - to further explain the action that is about to take place, add a supporting text to indicate the action.
-- IF **deposit** -> include in the text the new amount that is required and is about to be deposited 
-- IF **withdraw** ->  include in the text the amount that is removed (withdrawn) from the strategy allocation
- allocation and wallet amounts - the amount needs to reflect the latest state of the strategy according to user input or action performed
``` 
example:
allocation = 100
wallet = 500
user action: deposit 50
distribution section should immediately update the numbers to reflect this action
allocation displayed = 150
wallet displayed = 450
```
### Edit prices flow
This section will explain how to response based on user input or lack of

1. As you open the edit view
- enable interactive price chart area - allowing users to move the prices as we have on the create view.
- enable price inputs - allowing users to input specific prices as we have on the create view.
- enable spread change/input - allowing users to change the spread as we have on the create view.
- [NEW SECTION] show the first step of the **budget** flow: **Select token**
- disable the CTA
[default view mockup](https://www.figma.com/file/YpsatcY9jtxJr0SJwlFYNj/Carbon-Screen-Design?type=design&node-id=14733-68604&mode=design&t=X0hrFXBNX04yQUIA-4)

2. Following a user change on the price or spread IF no token is selected in the **budget: Select token** section
- expose a warning message at the **budget: Select token** section
[following edit mockup](https://www.figma.com/file/YpsatcY9jtxJr0SJwlFYNj/Carbon-Screen-Design?type=design&node-id=14733-68135&mode=design&t=X0hrFXBNX04yQUIA-4)

3. Following a user token selection
- expose **budget** step 2: **Edit budget**. this section should be default as closed.
- expose **budget** step 3: **Distribution**. automatically calculate the distribution based on the user selection. assuming the currently allocated budget of the selected token, is the budget to use for calculations.
- enable the CTA 
[mockup](https://www.figma.com/file/YpsatcY9jtxJr0SJwlFYNj/Carbon-Screen-Design?type=design&node-id=14733-65981&mode=design&t=X0hrFXBNX04yQUIA-4)

4. Following a user **Edit budget** action
- IF the user updated the budget using section 2 **Edit budget**, re-calculate the **distribution** using this input value and action
- IF there is no additional error (insufficient budget for example), enable the CTA
[mockup](https://www.figma.com/file/YpsatcY9jtxJr0SJwlFYNj/Carbon-Screen-Design?type=design&node-id=14733-67037&mode=design&t=X0hrFXBNX04yQUIA-4)

### LINKED ISSUE #1131: Overlapping strategy - update budget flow and design
prerequisite #948 

# Budget design update
This issue will outline a flow and design changes that will be applied on the deposit, withdraw and create **overlapping** strategy flows.

## Mockup
https://www.figma.com/file/YpsatcY9jtxJr0SJwlFYNj/Carbon-Screen-Design?type=design&node-id=14323-23752&mode=design&t=r7233fkw4bHiMa91-4

### Deposit (overlapping only)
Following the **exact** same flow as outlined in issue #948 with 1 difference:
- On step 2: **Edit budget**, remove the action toggle and only expose the option to **Deposit**
[deposit flow mockup](https://www.figma.com/file/YpsatcY9jtxJr0SJwlFYNj/Carbon-Screen-Design?type=design&node-id=14733-64884&mode=design&t=X0hrFXBNX04yQUIA-4)

### Withdrawal (overlapping only)
Following the **exact** same flow as outlined in issue #948 with 1 difference:
- On step 2: **Edit budget**, remove the action toggle and only expose the option to **Withdraw**
[withdrawal flow mockup](https://www.figma.com/file/YpsatcY9jtxJr0SJwlFYNj/Carbon-Screen-Design?type=design&node-id=14733-63761&mode=design&t=X0hrFXBNX04yQUIA-4)

### Create new strategy (overlapping only)
Following the **Deposit** flow (as new strategy always start with 0 budget and can only make a deposit), with the following difference:
- given allocation starts at 0, pay attention to the initial colors
[create overlapping strategy mockup](https://www.figma.com/file/YpsatcY9jtxJr0SJwlFYNj/Carbon-Screen-Design?type=design&node-id=14733-59911&mode=design&t=X0hrFXBNX04yQUIA-4)

### PR DESCRIPTION
**Pull request by GrandSchtroumpf**
Require #1140 to be merged for all warning to be WarningWithIcon

fix #948
fix #1131

### CODE REVIEW

**Comment by ashachaf** [comment]
- [ ] insufficient budget indication
when interacting with token0, there might be a case in which the result requires the user to deposit an amount from token1.
this amount needs to be checked against the token1 wallet balance and throw an "insufficient balance" error if needed.
image

mockup: TBA

- [ ] do not recalculate budgets unless prices have changed
flow:
- user enters the "edit price" page
- scroll to select anchor
expected: budget distribution should not indicate any deposit/withdraw needed
actual: 
image


- [ ] playing with prices lead to the allocation split to disappear 
image


- [ ] following a price edit, error is shown
Cannot read properties of undefined (reading 'toString')

- [ ] once anchor is selected, enable the CTA
currently:
image

**Comment by ashachaf** [comment]
- [ ] edit budget section
when closing the section (collapsing it), reset any input data that was entered by the user.

- [ ] when entering the "edit prices" page, hide "budget distribution" section until the user selects an anchor

- [ ] incorrect calculation
image

- [ ] optional - adjusting the prices or spread should reset the anchor selection

- [ ] edit budget - deposit check for wallet balance
when a user input deposit amount, there should be a check against current wallet budget
IF input amount > wallet balance:
- do not calculate distribution on the progress bars section
- indicate "insufficient funds" under the input field

current:
image

updated: (the screen shot is taken from the "deposit" into existing strategy flow)
image

- [ ] edit budget - withdraw check against allocation amount
when a user input withdraw amount, there should be a check against current order allocation budget
IF input amount > order allocation balance:
- do not calculate distribution on the progress bars section
- indicate "insufficient funds" under the input field

- [ ] progress bar - insufficient budget error
there are 2 flows with slightly different errors to them
- IF user input budget amount (in the `edit budget` section under deposit/withdraw) AND there is an insufficient balance indication on the 2nd budget needed
-- error message: "You should deposit {amount} {token} from your wallet, but your wallet has insufficient balance. Consider changing token deposit amount or prices."
- IF user update prices, spread or just select an anchor AND there is an insufficient balance indication on the progress bar
-- error message: "You should deposit {amount} {token} from your wallet, but your wallet has insufficient balance. Consider changing prices, token selection or budget."

- [ ] font size and weight are not aligned
this should be 12px and bold should be medium
image

File: src/components/common/imager/Imager.tsx [hunk 1]
```diff
@@ -76,6 +77,18 @@ export const Imager = ({
   );
 };
 
+export const TokenLogo = ({ token, size }: { token: Token; size: number }) => {
```
Comment (pingustar) [inline src/components/common/imager/Imager.tsx:80]: I think we need to ensure that logos are always a circle. Probably worth using `LogoImager`

File: src/components/strategies/overlapping/OverlappingBudgetDistribution.tsx [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+          <span className="text-white">
+            {tokenAmount(initialBudget, token)}
+          </span>
+        </label>
+        <label htmlFor={walletId}>
+          Wallet:&nbsp;
+          <span className="text-white">{tokenAmount(balance, token)}</span>
+        </label>
+      </div>
+      <div className={styles.progress}>
```
Comment (pingustar) [inline src/components/strategies/overlapping/OverlappingBudgetDistribution.tsx:62]: why not use tailwind classes?

File: src/pages/strategy/edit/overlapping/index.tsx [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export const EditOverlappingStrategyPage = () => {
+  return <></>;
```
Comment (pingustar) [inline src/pages/strategy/edit/overlapping/index.tsx:2]: what is this needed for?

**Review by pingustar — COMMENTED** [review]
@tiagofilipenunes @GrandSchtroumpf there is a lot of file changes that are only tailwind class order/sort changes. Is this due to the new tailwind version? Lets apply these changes in a dedicated PR - so it wont add so much noise in new PRs.

@GrandSchtroumpf I am looking forward to the `form` presentation ;) I still feel like the forms make things more complicated. For example using a ref on a form, then checking for error classes and then a useEffect for input validation - I feel like this could just be derived from state without the need for a ref / effect.

File: src/components/strategies/overlapping/OverlappingBudgetDistribution.tsx (hunk 2 above)
Comment (GrandSchtroumpf) [reply src/components/strategies/overlapping/OverlappingBudgetDistribution.tsx:62]: In an early version I was using some complex CSS selector and keyframes animations. We simplified everything to use only flex-grow so I think I can move it back to Tailwind indeed👍

File: src/pages/strategy/edit/overlapping/index.tsx (hunk 3 above)
Comment (GrandSchtroumpf) [reply src/pages/strategy/edit/overlapping/index.tsx:2]: An artifact from an old time when a dev thought he could use this PR as a base for a better place where each strategy type would have its own page. The delusional dev soon realized that it was a chimera and, in front of the complexity lying before him, decided to focus his energy into the sole task of editing overlapping strategies. In his journey he forgot the ground is initially prepared.

Hopefully you were here to ask him to clean his shit XD

File: src/components/strategies/edit/overlapping/EditBudgetOverlappingStrategy.tsx [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+  const spread = getRoundedSpread(strategy);
+
+  const newMarketPrice = useMarketPrice({ base, quote });
+  const initialMarketPrice = geoMean(
+    strategy.order0.marginalRate,
+    strategy.order1.marginalRate
+  )!;
+  const marketPrice = touched
+    ? newMarketPrice.toString()
+    : initialMarketPrice!.toString();
```
Comment (tiagofilipenunes) [inline src/components/strategies/edit/overlapping/EditBudgetOverlappingStrategy.tsx:56]: Throwing error when editing strategy with empty budget

File: src/components/strategies/overlapping/OverlappingBudget.module.css [hunk 5]
```diff
@@ -0,0 +1,18 @@
+/* TODO(#1130) integrate this into component when tailwind is updated */
```
Comment (tiagofilipenunes) [inline src/components/strategies/overlapping/OverlappingBudget.module.css:1]: Can this be integrated into the component now?

**Review by tiagofilipenunes — CHANGES_REQUESTED** [review]
Great work!

- Strategy Creation with 0 budget shows a warning modal about creating an empty strategy 👍
- Very clear and beautiful UX 🔥

Notes:
- [ ] Tooltip icons on create overlapping strategy have h-14 w-14 whereas on other create strategy forms like limit/recurring, they have h-18 w-18
- [ ] Creating recurring strategy with more budget than available in wallet, which shows the error "Insufficient balance" does not disable CTA.
- [ ] Creating recurring strategy with wallet disconnected showing the button text "Connect wallet" has disabled CTA. On overlapping strategy, "Insufficient Balance" is shown and CTA is disabled.
- [ ] Can you please add `scaleUp` to animation.animate in src/libs/twmerge/index.ts? To make sure the merge is correct - [ ] tailwind merge doesn't read the tailwind.config.ts. If working correctly, adding this unit test in twmerge.test.ts should pass:
```ts

    it('should merge different non-default animations', () => {
      expect(customTwMerge('animate-spin animate-scaleUp')).toEqual(
        customTwMerge('animate-scaleUp')
      );
    });
```
- [ ] In edit price, the MarketPriceIndication text shows yellow for min buy price below market price. Is this expected as it suggests a warning?
- [ ] Since edit is now available, can you uncomment lines 130-135 of useCreateStrategyTypeMenu?
- [ ] OverlappingBudgetDescription: icon an text are not centered vertically which is the same as the design. Imo looks better aligned center
- [ ] Deposit option not available on overlapping strategy manage button for strategies with no budget (StrategyBlockManage).

Error: initialMarketPrice is undefined
Flow:
1. Create overlapping strategy with empty budget
2. Go to strategy manage button > "Edit Prices"
3. Error `initialMarketPrice is undefined` shows up

File: src/components/strategies/create/overlapping/CreateOverlapping.tsx [hunk 6]
```diff
@@ -0,0 +1,404 @@
...
+  })();
+
+  const marketPrice = useMarketPrice({ base, quote });
+
+  const calculateBuyBudget = (
+    sellBudget: string,
+    buyMin: string,
+    sellMax: string
+  ) => {
+    if (!base || !quote) return;
```
[the remark this answers is not in the document]
Comment (GrandSchtroumpf) [reply src/components/strategies/create/overlapping/CreateOverlapping.tsx]: They can be undefined at initialization

File: src/components/strategies/edit/overlapping/EditBudgetOverlappingStrategy.tsx (hunk 4 above)
Comment (GrandSchtroumpf) [reply src/components/strategies/edit/overlapping/EditBudgetOverlappingStrategy.tsx:56]: Good catch

**Comment by Costa769** [comment]
A few notes:


[ ] Overlapping ->deposit->Incorrect calculation for the deposit amount of the second token
image

[ ] Overlapping->deposit-> when there is not enough tokens in the wallet to proceed with the deposit and the error message is shown, CTA is still enabled

[ ] Overlapping->deposit-> 0 input enables the CTA
image

[ ] Recurring strategy->withdraw->clicking on ‘Max” does not fill the withdraw input field with the allocated amount(but it enables the CTA and allows user to  proceed with withdrawing funds
image

[ ] Inactive overlapping strategy without budget-> edit prices-> when user clicks on edit prices in the menu, he lands on an error page
image

[ ] Inactive overlapping strategy without budget->menu->there is no duplicate and deposit options in the menu
image

[ ]

**Comment by ashachaf** [comment]
- [ ] edit overlapping strategy - "in the money" warning
currently: this warning only shows up AFTER the user changes the prices of the strategy.
however, this flow can result in cases where the strategy is "out of the money" but there is no warning.
update: when opening the edit overlapping strategy, IF strategy is out of money (use the existing check there is), show the relevant warnings (before any user interaction).

https://github.com/bancorprotocol/carbon-app/assets/40430285/8f81b5ad-33e3-4335-90d4-f531eb0d4db0

- [ ] add checkbox 
currently: when a user update prices, there is an indication to choose token. following that, the CTA is enabled while the transaction might result in deposit/withdraw as part of the process.
updated: add a checkbox (same as we have when there is a warning) that forces the user to indicate they understand the changes they make to the strategy.
text: I've approved the edits and distribution changes.
_this should be applied on edit and create overlapping strategy_

- [ ] recalculate distribution
please add and apply this on the input field.
if user input a value into the withdraw/deposit input field, apply recalculation for the distribution only if input value > 0.

- [ ] text change
current: Please specify which token you'd prefer to use as the anchor.
updated: Please select a token to proceed.

- [ ] deposit flow
issue we identify: when depositing 0 amount, the recalculated distribution of the strategy indicates that some deposit is needed.
flow:
1. click to deposit into an overlapping strategy
2. select base token
3. input 0
4. see incorrect recalculation of budgets
5. proceed
6. repeat step 1-3 on the same strategy, this time the recalculation will not show any need for new funds.
i believe this indicates that the incorrect recalculation is caused due to incorrect value used in the deposit process.

- [ ] tool tip for the budget section
image
text: Indicate the token, action and amount for the strategy. Note that in order to maintain the overlapping behavior, the 2nd budget indication will be calculated using the prices, spread and budget values you use.

**Comment by ashachaf** [comment]
- [ ] diff from market price 
if the buy low price is below market price, it should NOT be orange.
if the sell high price is above market price, it should NOT be orange.
image

reference: https://github.com/bancorprotocol/carbon-app/issues/1120

**Comment by ashachaf** [comment]
- [ ] edit price - default market price
when entering the "edit price" view of an overlapping strategy:
- chart should show the current (external/user defined) market price
- buy/sell distribution should use the strategy details based on their current details
once a user update anything, we update the strategy to use the current (external/user defined) market price and not the previous strategy price.

- [ ] edit price - lead to deposit/withdraw should also update the z value
flow:
- create an overlapping strategy where market price is within the selected range and add funds to both sides
- (at this stage, the z value is calculated based on range, budget and market/marginal price)
- edit prices for this strategy and move it to be above or below the market price. this will result in withdrawing the full budget of one order.
expected: z value to be re-calculate for both orders given the new range, budget and market/marginal price.
actual: 
- on the overview card, the "empty" order seems to have z value = budget and the order with budget maintain previous marginal price. 
- when editing prices again, the z value (for both orders) seems to be according to the previous settings.
image
image

**Comment by ashachaf** [comment]
text changes:
- [ ] withdraw flow
under "Distribution"
current:
Following the edits implemented above, these are the changes in budget allocation
updated:
Following the above withdraw amount, these are the changes in budget allocation

- [ ] deposit flow
under "Distribution"
current:
Following the edits implemented above, these are the changes in budget allocation
updated:
Following the above deposit amount, these are the changes in budget allocation

**Comment by ashachaf** [comment]
- [ ] deposit flow
add the chart at the top of the page.
chart is not interactive and serve as an indicator to the current strategy details.
chart will display **current** strategy details.
[mockup](https://www.figma.com/design/YpsatcY9jtxJr0SJwlFYNj/Carbon-Screen-Design?node-id=14733-64885&t=iQofe5vHDmyGypLX-4)

- [ ] withdraw flow
add the chart at the top of the page.
chart is not interactive and serve as an indicator to the current strategy details.
chart will display **current** strategy details.
[mockup](https://www.figma.com/design/YpsatcY9jtxJr0SJwlFYNj/Carbon-Screen-Design?node-id=14733-63762&t=iQofe5vHDmyGypLX-4)

- [ ] text updates
on the strategy type selection, when choosing "overlapping" please add these benefits under the existing text
current:
Overlapping Liquidity
A concentrated position where you buy and sell in a custom price range, used to create a bid-ask spread that moves as the market does

updated:
Overlapping Liquidity
A concentrated position where you buy and sell in a custom price range, used to create a bid-ask spread that moves as the market does

Adjustable
Easily edit your price range and position size without having to withdraw and redeposit into a new position, saving you time, gas and a whole lotta headache.

Auto-compounding profits
Your profits stay within your position, earning you more!

- [ ] text change
on "create" overlapping strategy, we show a checkbox with the following text
current: "I've approved the edits and distribution changes."
updated: "I've approved the token deposit(s) and distribution."

- [ ] edit prices + deposit flow
- edit an existing strategy and move it below market price
- select the buylow token (quote) and deposit any amount that you have in the wallet
expected:
- sell high budget would indicate a full withdraw as the strategy is now below market price
- deposit input amount validation should be against wallet balance and let the user proceed
actual:
https://github.com/bancorprotocol/carbon-app/assets/40430285/b7c2be28-2c69-4c75-866c-1e4f17d773a3

- [ ] (optional) update checkbox text when choosing anchor token that has 0 budget
flow:
- user edit prices of a strategy
- select anchor token to use as the value for calculating the token budget distribution
current:
IF selected token budget = 0, we calculate the 2nd token and eventually withdraw the full budget from the strategy with no special warning
updated:
IF selected token budget = 0, 
- update the checkbox text
- text: I've approved the edits and distribution changes and aware that the strategy will be empty and inactive. 
image

- [ ] (optional) expose manage menu in strategy page
currently: 
open strategy page of a strategy you own and see the short menu
image

update:
open strategy page of a strategy you own and see the full manage menu
image

- [ ] edit prices: incorrect "insufficient" message
edit price flow, select token and try to withdraw. 
expected: IF the input amount > allocated budget -> show the warning below the input field
image

File: src/components/strategies/create/CreateOverlapping.tsx [hunk 7]
```diff
@@ -0,0 +1,453 @@
...
+  OverlappingMarketPrice,
+} from '../overlapping/OverlappingMarketPrice';
+import { UserMarketPrice } from '../UserMarketPrice';
+
+interface Props {
+  base: Token;
+  quote: Token;
+  order0: OrderCreate;
+  order1: OrderCreate;
+  // TODO Remove spread from useCreateStrategy and use it only locally
```
Comment (tiagofilipenunes) [inline src/components/strategies/create/CreateOverlapping.tsx:43]: Can you create an issue for this or remove please?

**Review by tiagofilipenunes — CHANGES_REQUESTED** [review]
Great work again! Just some observations. Didn't look at the e2e code, only the changes from last review.

- [ ] When editing prices on an overlapping strategy with a token with unknown market price, when entering the price, two warning checkboxes are shown but the one to enable the CTA probably shouldn't show while the price is being set by the user.

image

- [x] Creating an empty overlapping strategy doesn't show modal with warning about creating empty strategy like recurring does
- [x] Using input "." in enter market price form field updates to value "NaN"
image

- [ ] When edit prices of PARQ/USDC, going to budget and deposit is showing "Insufficient Balance" with input 0

image

- [ ] Is it worth it to show the distribution on the simulate page? It will always show two graphs with 50% area occupied for the allocation and balance if the values are valid. Meaning it doesn't add any information to the user 

image

Can you replace the animation object in the customTwMerge config with this one please?

```typescript

      animation: [
        {
          animate: [
            'none',
            'spin',
            'ping',
            'pulse',
            'bounce',
            'slideUp',
            'scaleUp',
          ],
        },
      ],
```
- [ ] Regression: When wallet is not connected, on create [overlapping strategy](http://localhost:3000/strategies/create?base=0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE&quote=0x6B175474E89094C44Da98b954EedeAC495271d0F&strategyType=recurring&strategySettings=overlapping) and create [recurring strategy](strategies/create?base=0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE&quote=0x6B175474E89094C44Da98b954EedeAC495271d0F&strategyType=recurring&strategySettings=range), the connect wallet CTA shows disabled for me. 

image

- I get an error when inputting '.' for both the overlapping creation/edit prices as well as the simulator, but I assume that can be covered by #1218

File: src/components/strategies/overlapping/useOverlappingMarketPrice.ts [hunk 8]
```diff
@@ -0,0 +1,17 @@
+import { useMarketPrice } from 'hooks/useMarketPrice';
+import { Strategy } from 'libs/queries';
+import { geoMean } from 'utils/fullOutcome';
+
+export const hasNoBudget = (strategy: Strategy) => {
+  const { order0, order1 } = strategy;
+  return !Number(order0.balance) && !Number(order1.balance);
```
[the remark this answers is not in the document]
Comment (GrandSchtroumpf) [reply src/components/strategies/overlapping/useOverlappingMarketPrice.ts:7]: SafeDecimal("0") won't be NaN, so it wouldn't work.

**Review by ashachaf — APPROVED** [review]

**Pull request merged**